File: src/main/java/vip/yazilim/libs/springcore/exception/DatabaseReadException.java
package vip.yazilim.libs.springcore.exception;

import java.util.Arrays;

public class DatabaseReadException extends DatabaseException {
    private static final long serialVersionUID = -3490767649996233498L;

    public <E, ID> DatabaseReadException(Class<E> entityClass, Exception exception) {
        super(entityClass, "READ", "", exception);
    }

    public <E, ID> DatabaseReadException(Class<E> entityClass, Exception exception, Object... params) {
        super(entityClass, "READ", Arrays.toString(params), exception);
    }
}

File: src/main/kotlin/vip/yazilim/libs/springvip/util/generic/service/exception/DatabaseReadException.kt
package vip.yazilim.libs.springvip.util.generic.service.exception

import kotlin.reflect.KClass

// READ
class DatabaseReadException : DatabaseException {
    constructor(entityClass: KClass<*>, id: Any, exception: Exception)
            : super(entityClass, "READ", id.toString(), exception)

    constructor(entityClass: KClass<*>, exception: Exception, vararg params: Any?)
            : super(entityClass, "READ", params.contentToString(), exception)
}